### PR TITLE
Add downloadUrlYear() to Download Charge Log for Year.

### PR DESCRIPTION
**Pull request by Sleepwalker86**

### CODE REVIEW

File: public/downloadChargeLog.php [hunk 1]
```diff
@@ -32,15 +32,29 @@
...
+if (!preg_match("/[0-9]{4}/", $_GET["year"])) {
+    http_response_code(400);
+    die("invalid data");
 }
 
-$file_name = sprintf('%04d%02d', $_GET["year"], $_GET["month"]);
-$charge_log_file = $charge_log_path . $file_name . ".json";
-$charge_log_data = json_decode(file_get_contents($charge_log_file), true);
-
+// Wenn der Monat angegeben ist und gültig, dann verwenden wir diesen Monat
```
Comment (benderl) [inline public/downloadChargeLog.php:40]: Please use English vor comments in the code.

File: public/downloadChargeLog.php [hunk 2]
```diff
@@ -32,15 +32,29 @@
...
+if (isset($_GET["month"]) && preg_match("/^((0?[1-9])|(1[0-2]))$/", $_GET["month"])) {
```
Comment (benderl) [inline public/downloadChargeLog.php:41]: This modification lacks the check for invalid data on `$_GET["month"]`. If "month" is set and the regex does not match, there should be an `http error 400`.

File: src/views/ChargeLog.vue [hunk 3]
```diff
@@ -489,6 +490,11 @@ export default {
       const baseUrl = `${location.protocol}//${location.hostname}:${port}/openWB/web/settings/downloadChargeLog.php`;
       return baseUrl + `?year=${this.chargeLogRequestData.year}&month=${this.chargeLogRequestData.month}`;
     },
+    downloadUrlYear() {
+      const port = parseInt(location.port) || (location.protocol == "https:" ? 443 : 80);
+      const baseUrl = `${location.protocol}//${location.hostname}:${port}/openWB/web/settings/downloadChargeLog.php`;
```
Comment (benderl) [inline src/views/ChargeLog.vue:495]: As we have two URLs now, the property `downoadURL()` should be renamed to something more meaningful like `downloadUrlMonth()`.

Please do not duplicate code. Create a new computed property for the `baseURL` and make use of it in `downloadUrlMonth()` and `downloadUrlYear()`.

**Review by benderl — CHANGES_REQUESTED** [review]

**Review by benderl — APPROVED** [review]

File: src/views/ChargeLog.vue (hunk 3 above)
Comment (benderl) [reply src/views/ChargeLog.vue:496]: @Sleepwalker86 just a reminder

**Comment by Sleepwalker86** [comment]
i hope now is all fine

File: src/views/ChargeLog.vue [hunk 4]
```diff
@@ -484,11 +486,24 @@ export default {
...
+      }
+      return `${this.baseUrl}?year=${this.chargeLogRequestData.year}&month=${this.chargeLogRequestData.month}`;
+    },
+    downloadUrlYear() {
+      if (!this.chargeLogRequestData.year) {
+        console.error("Fehlendes Jahr");
+        return null; // oder ein Standardwert
+      }
+      return `${this.baseUrl}?year=${this.chargeLogRequestData.year}`;
+    }
```
Comment (benderl) [inline src/views/ChargeLog.vue:506]:
```suggestion
    },
```

**Pull request merged**